sdk/core/corehttp/corehttp/runtime/policies/_authentication.py

```diff
@@ -142,7 +145,9 @@ def send(self, request: PipelineRequest[HTTPRequestType]) -> PipelineResponse[HT
         :return: The pipeline response object
         :rtype: ~corehttp.runtime.pipeline.PipelineResponse
         """
-        self.on_request(request, auth_flows=self._auth_flows)
+        op_auth_flows = request.context.options.pop("auth_flows", None)
+        auth_flows = op_auth_flows if op_auth_flows is not None else self._auth_flows
+        self.on_request(request, auth_flows=auth_flows)
```

Comment:
Could you confirm that if credential is not provided in the client ctor, auth policy is not added into the pipeline?

Reply:
The generated code always has `if self.credential and not self.authentication_policy:` guarding the creation of the auth policy, so if no credential is provided, no auth policy will be added to the pipeline.